### PR TITLE
[ES-9906] Add generic shopping assistant widget

### PR DESCRIPTION
**Pull request by wkurniawan07**

### CODE REVIEW

File: src/official-widgets/shopping-assistant/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+## Local development
+
+1. To run the widget locally:
+   1. Add your app key and placement ID to `dev-configs.ts`.
+   2. Run:
+      ```sh
+      npm run start:shopping-assistant
+      ```
+      The dev server will be available at `http://localhost:8080` and will automatically reload for changes made in `src/official-widgets/shop-the-look` folder.
```
Comment (shonshon99) [inline src/official-widgets/shopping-assistant/README.md]: should be `src/official-widgets/shopping-assistant`

File: src/official-widgets/shopping-assistant/components/ProductCard.tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  return (
+      <a href={product.product_url} ref={targetRef} onClick={() => {
+        productSearch.send(Actions.PRODUCT_CLICK, {
+          label: 'dialogue',
+          pid: product.pid,
+          pos: index + 1,
+          queryId,
+          cat: 'Result',
+        });
+      }}>
```
Comment (shonshon99) [inline src/official-widgets/shopping-assistant/components/ProductCard.tsx]: Need to disable `onClick` and `href` if `debugMode` is true

File: src/official-widgets/shopping-assistant/index-dev.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+};
+
+const getRenderElement = (configs: WidgetConfig): HTMLDivElement => {
+  const { cssSelector, cameraButtonSelector } = configs.displaySettings;
+  return (
+      document.body.querySelector(cssSelector || `.ps-widget-${configs.appSettings.placementId}`)
+      || document.body.querySelector(`#${cameraButtonSelector}`)
+      || createElement(cameraButtonSelector)
+  );
+};
```
Comment (shonshon99) [inline src/official-widgets/shopping-assistant/index-dev.tsx]: Remove `cameraButtonSelector`

File: src/official-widgets/shopping-assistant/index.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+};
+
+const getRenderElement = (configs: WidgetConfig): HTMLDivElement => {
+  const { cssSelector, cameraButtonSelector } = configs.displaySettings;
+  return (
+      document.body.querySelector(cssSelector || `.ps-widget-${configs.appSettings.placementId}`)
+      || document.body.querySelector(`#${cameraButtonSelector}`)
+      || createElement(cameraButtonSelector)
+  );
+};
```
Comment (shonshon99) [inline src/official-widgets/shopping-assistant/index.tsx]: Remove `cameraButtonSelector`

**Review by Niflnir — CHANGES_REQUESTED** [review]

**Review by Niflnir — APPROVED** [review]
LGTM

**Pull request merged**